testing/cli-e2etest/testscenarios/datastore/list_datastore_test.go

```diff
@@ -84,12 +84,13 @@ func TestListDatastore(t *testing.T) {
 		result := tracetestcli.Exec(t, "list datastore --sortBy name --output json", tracetestcli.WithCLIConfig(cliConfig))
 		helpers.RequireExitCodeEqual(t, result, 0)
 
-		dataStoresJSON := helpers.UnmarshalJSON[[]types.DataStoreResource](t, result.StdOut)
+		dataStoresList := helpers.UnmarshalJSON[types.ResourceList[types.DataStoreResource]](t, result.StdOut)
```

Comment:
this scenario was incorrect (as well as the old implementation). it was returning an array of resources instead of a `ResourceList`